### ISSUE TITLE
Make sure all things are cleanly runnable offline

### ISSUE THREAD

**Issue by aronchick** [filecoin-project/bacalhau#1029]
ETA: 2023Q1 

Repro:

```
make devstack --node 1 # Terminal 1
go run . ubuntu echo hello world # Terminal 2 <- requires having run it locally before
```

Observe:
```
2022/11/07 11:49:47 max retry time elapsed: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp: lookup api.honeycomb.io: i/o timeout"
```

And many other. Let's fix this.

**Comment by wjam** [comment]
The workaround for the issue in the logs is to _unset_ the `HONEYCOMB_KEY` environment variable, as that is making Bacalhau export traces to Honeycomb.

The real problem, which will become more of an issue when others are running Bacalhau servers themselves, is that Bacalhau isn't flexible with where to output traces to. This will be more of an issue with other people as they may want tracing information to go to other provides - Lightstep, Grafana Tempo, etc.

I would argue that supporting the different configuration needed for these OTEL destinations would be beyond what can be accomplished with just command line arguments (e.g. arbitrary number of headers to add to the trace exporter), and so would need some sort of configuration file.

We would also need to move away from relying on the tracer being set to some global variables, as the tracing can only be configured once the command flags are parsed, to storing the tracer inside the `context.Context`. This has the advantage of removing globals and so making it easier to run tests in parallel (see #1034). A similar move will need to be done with logging - using `log.Ctx` to retrieve a preconfigured logger rather than relying on the static on at `github.com/rs/zerolog/Logger` - so that we can run the tests in parallel and see log output.

**Comment by wjam** [comment]
I'm able to run `./bin/darwin_amd64/bacalhau docker run ubuntu echo hello world` against a local devstack without internet connection when `HONEYCOMB_KEY` isn't set - assuming `ubuntu:latest` is already cached. Are there any other examples of commands that fail, or is this mainly going to be about changing how we configure tracing?

**Comment by lukemarsden** [comment]
@aronchick can you comment on whether unsetting `HONEYCOMB_KEY` is sufficient to run offline, and if not what the next issue you see is? Of course, you will have to have docker images and such cached...

**Comment by lukemarsden** [comment]
as I said in #1008, 

Ideally, we make make test totally 100% self sufficient, only depending on golang (no network, no docker), and then we don't need this.

Leaving this issue open to track that effort

**Comment by aronchick** [comment]
This totally works - but setting HONEYCOMB_KEY feels like a bad fit (very undiscoverable). You tell me.

**Comment by wjam** [comment]
@aronchick The problem isn't that `HONEYCOMB_KEY` was unset, the problem is that it was set. Although I feel a better way would be to have a proper configuration file to handle configuring the trace destination (which I'm doing as part of #1153)

**Comment by wjam** [comment]
This appears to now be a duplicate of #635?
